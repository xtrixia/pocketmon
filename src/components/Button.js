import { css } from "@emotion/react";

import { SPACINGS } from "../root/spacings";
import { COLORS } from "../root/colors";

function Button({ children, fullWidth, type, ...props }) {
  let borderStyle = `1px solid ${COLORS.dark}`;
  let backgroundStyle = COLORS.white;
  let boxShadowStyle = "1px 2px";
  if (type === "outline") {
    borderStyle = "none";
    backgroundStyle = "none";
    boxShadowStyle = 0;
  }

  return (
    <button
      className={css`
        width: ${fullWidth ? "100%" : "fit-content"};
        padding: ${SPACINGS.xs};
        background: ${backgroundStyle};
        box-shadow: ${boxShadowStyle};
        border-radius: 5px;
        border: ${borderStyle};
      `}
      {...props}
    >
      {children}
    </button>
  );
}
export default Button;
