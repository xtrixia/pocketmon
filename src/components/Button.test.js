import React from "react";
import renderer from "react-test-renderer";

import Button from "./Button";

test("Button rendered with unique id passed from props", () => {
  const component = renderer.create(
    <Button id="btn-click-here">Click here</Button>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.props).toHaveProperty("id", "btn-click-here");
});

test("Button onClick function works as expected", () => {
  const mockFn = jest.fn();
  const a = new mockFn();

  const component = renderer.create(
    <Button id="btn-click-here" onClick={mockFn}>
      Click here
    </Button>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.props.onClick();
  expect(mockFn.mock.instances[0]).toBe(a);
});
