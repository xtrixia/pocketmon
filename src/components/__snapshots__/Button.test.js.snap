// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button onClick function works as expected 1`] = `
<button
  className="css-14uyppl"
  id="btn-click-here"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Click here
</button>
`;

exports[`Button rendered with unique id passed from props 1`] = `
<button
  className="css-14uyppl"
  id="btn-click-here"
>
  Click here
</button>
`;
